terraform/modules/service/variables.tf

```diff
@@ -0,0 +1,153 @@
+variable "cluster" {
+  description = "The ecs cluster hosting the service and task."
+  type        = any
+}
+
+variable "container_environment" {
+  description = "The environment variables to pass to the container"
+  type = list(object({
+    name  = string
+    value = string
+  }))
+  default = null
+}
+
+variable "container_secrets" {
+  description = "The secrets to pass to the container. For more information, see [Specifying Sensitive Data](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/specifying-sensitive-data.html) in the Amazon Elastic Container Service Developer Guide"
+  type = list(object({
+    name      = string
+    valueFrom = string
+  }))
+  default = null
+}
+
+variable "cpu" {
+  description = "Number of cpu units used by the task."
+  type        = number
+}
+
+variable "desired_count" {
+  description = "Number of instances of the task definition to place and keep running."
+  type        = number
+  default     = 0
+}
+
+variable "family_name_override" {
+  default     = null
+  description = "The desired family name for the ECS task definition.  If null will default to: {var.platform.env}-{var.platform.app}-{var.platform.service}"
+}
+
+variable "force_new_deployment" {
+  default     = false
+  description = "When *changed* to `true`, trigger a new deployment of the ECS Service even when a deployment wouldn't otherwise be triggered by other changes. **Note**: This has no effect when the value is `false`, changed to `false`, or set to `true` between consecutive applies."
+  type        = bool
+}
+
+
+variable "image" {
+  description = "The image used to start a container. This string is passed directly to the Docker daemon. By default, images in the Docker Hub registry are available. Other repositories are specified with either `repository-url/image:tag` or `repository-url/image@digest`"
+  type        = string
+  default     = null
+}
```

Comment:
I think the image tag should be required. Is there a use case for having a service and task without an image?